### PR TITLE
calc

### PR DESCRIPTION
**Pull request by DzmitryKosmach**

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+function multiply(x, y) {
+  return x * y;
+}
+
+function divide(x, y) {
+  return x / y;
+}
+
+function divideInteger(x, y) {
+  return ((x - x % y) / y);
```
Comment (NikitaSedyx) [inline js/main.js]: nit: it will work w/o ()
`return (x - x % y) / y;`

File: js/main.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+var values = [[1, "rt"], [2, true], [3, false], [4, 1], [5, {}], [6, "2"], [7], []];
+
+autoCalc();
+
+function autoCalc() {
+  for (i = 0; i < values.length; i++) {
```
Comment (NikitaSedyx) [inline js/main.js]: do you remember func `forEach`? I'd suggest to use it, it looks much better

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    logAction(x, y, add);
+    logAction(x, y, sub);
+    logAction(x, y, multiply);
+    logAction(x, y, divide);
+    logAction(x, y, divideInteger);
+    console.log('\n');
+  }
+}
+
+function logAction(x, y, func) {
```
Comment (NikitaSedyx) [inline js/main.js]: I hope you remember and understand mechanism of hoisting and reason you can use  this func in `autoCalc`

**Review by NikitaSedyx — APPROVED** [review]
you are able to merge, but if you will change code according my suggestions please wait another review ,thanks

**Comment by DzmitryKosmach** [comment]
Привет!
Базовые функции калькулятора (commit: foreach):

https://github.com/DzmitryKosmach/DzmitryKosmach.github.io/pull/1/commits


2017-03-23 21:03 GMT+03:00 Mikita Siadykh <notifications@github.com>:

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+var values = [[1, "rt"], [2, true], [3, false], [4, {}], [5, 3], [6, "4"], [7], []];
+
+values.forEach(autoCalc);
```
Comment (NikitaSedyx) [inline js/main.js:5]: much better

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**